pkg/applicationserver/io/packages/redis/pagination.go

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redis
+
+import (
+	"context"
+
+	"github.com/go-redis/redis"
+	"go.thethings.network/lorawan-stack/pkg/rpcmetadata"
+)
+
+type paginationOptionsKeyType struct{}
```

Comment:
Can we promote this to `pkg/redis`?

And how is the total value returned?

Reply:
The total value is returned by the call itself through headers (metadata / HTTP headers). It follows the same structure as the Identity Server.

I think it can be promoted to `pkg/redis`. @htdvisser thoughts on this ?

Reply:
I agree with @johanstokking (this thread) and @rvolosatovs (https://github.com/TheThingsNetwork/lorawan-stack/pull/1399#discussion_r331949623)